File: javascript/1164.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split("\n");
var q = parseInt(lines.shift());

for(let i = 1; i <= q; i++){
    var num = parseInt(lines.shift());
    var perfeitos = [];
    var sum = 0;
    for(let j = 1; j < num; j++){
        if(num % j === 0){
            perfeitos.push(j);
        }
    }
    var lg = perfeitos.length
    for(let j = 0; j < lg; j++){
        sum += perfeitos[j];
    }
    console.log(sum == num ? `${num} eh perfeito` : `${num} nao eh perfeito`);
}

File: C#/1164.cs
using System; 

class URI {

    static int ehPerfeito(int n){
        int sum = 0;
        for(int i = 1; i < n; i++){
            if(n % i == 0){
                sum += i;
            }
        }
        if(sum == n) return 1;
        return 0;
    }
    
    static void Main(string[] args) { 
        int i, num, n = int.Parse(Console.ReadLine());
        
        for(i = 0; i < n; i++) {
            num = int.Parse(Console.ReadLine());
            if(ehPerfeito(num) == 1){
                Console.WriteLine(num + " eh perfeito");
            }else {
                Console.WriteLine(num + " nao eh perfeito");
            }
        }
    }
}